### ISSUE TITLE
Victorialogs can't parse syslog timestamps

### ISSUE THREAD

**Issue by hasso** [VictoriaMetrics/VictoriaMetrics#6508]
### Describe the bug

Simple UDP syslog receiver doesn't receive any message.

`Jun 18 23:37:20 syslog-server victoria-logs-prod[59930]: 2024-06-18T23:37:20.331Z        error        VictoriaMetrics/app/vlinsert/syslog/syslog.go:252        syslog: cannot process UDP data from 192.168.1.120:48789 at 192.168.1.1:514: cannot read line #0: cannot get timestamp from syslog line "<10>1 2024-06-18T23:37:20+00:00 sender tester - - [log sequenceId=\"1\"] this is a log message": cannot unmarshal iso8601 timestamp from timestamp="2024-06-18T23:37:20+00:00"`

### To Reproduce

It's a simple `victoria-logs-prod -syslog.listenAddr.udp=217.146.65.55:514` test.

### Version

```
$ ./victoria-logs-prod --version
victoria-logs-20240617-222537-tags-v0.20.2-victorialogs-0-g3eda4617c0
```

**Comment by lujiajing1126** [comment]
We also suffered from this issue, `2024-06-18T23:37:20+08:00` with timezone cannot be parsed properly

**Comment by lammel** [comment]
We ran into the same issue, seems like parsing using `insertutils.ExtractTimestampRFC3339NanoFromFields` is not working as expected.

From [RFC5424](https://www.rfc-editor.org/rfc/rfc5424#section-6.2.3) the Syslog messages header uses this ABNF:

```
      TIMESTAMP       = NILVALUE / FULL-DATE "T" FULL-TIME
      FULL-DATE       = DATE-FULLYEAR "-" DATE-MONTH "-" DATE-MDAY
      DATE-FULLYEAR   = 4DIGIT
      DATE-MONTH      = 2DIGIT  ; 01-12
      DATE-MDAY       = 2DIGIT  ; 01-28, 01-29, 01-30, 01-31 based on
                                ; month/year
      FULL-TIME       = PARTIAL-TIME TIME-OFFSET
      PARTIAL-TIME    = TIME-HOUR ":" TIME-MINUTE ":" TIME-SECOND
                        [TIME-SECFRAC]
      TIME-HOUR       = 2DIGIT  ; 00-23
      TIME-MINUTE     = 2DIGIT  ; 00-59
      TIME-SECOND     = 2DIGIT  ; 00-59
      TIME-SECFRAC    = "." 1*6DIGIT
      TIME-OFFSET     = "Z" / TIME-NUMOFFSET
      TIME-NUMOFFSET  = ("+" / "-") TIME-HOUR ":" TIME-MINUTE
```

`ExtractTimestampRFC3339NanoFromFields` currently only parses the TIME-OFFSET with a leading "Z", the `+/-` variants are not supported by that function.
From the tests this seems to be intentional as there are explicit tests that are expected to return failure with `+08:00` timezone format in `lib/logstorage/values_encoder_test.go`:

```
func TestTryParseTimestampRFC3339Nano_Failure(t *testing.T) {
        // ...
        // timestamp with timezone
	f("2023-01-16T00:45:51+01:00")
	f("2023-01-16T00:45:51.123+01:00")
```

**Comment by lammel** [comment]
Actually there is a comment by @valyala describing that there is an issue with reversing the timezone:

```
func TryParseTimestampRFC3339Nano(s string) (int64, bool) {
	// Do not parse timestamps with timezone other than Z, since they cannot be converted back
	// to the same string representation in general case.
	// This may break search.
```

If the timestamp is converted to nsecs for UTC, there should be no problem (other than not being able to 100% reproduce the timestamp string). Maybe he can shed some light, why this is important in this case.

**Comment by lammel** [comment]
For us this seems to only be an issue when using rsyslog forwarding using the RFC5424 format (called `RSYSLOG_SyslogProtocol23Format`)

```
*.* @@127.0.0.1:10514                                 # <= works using traditional format
*.* @@127.0.0.1:10514;RSYSLOG_SyslogProtocol23Format  # <= not working due to timestamp parsing error
```

**Comment by valyala** [comment]
The issue must be fixed in the commit 9e1c03724907649f8347f9f2752d200b196f6f1e , which will be included in the next VictoriaLogs release.

**Comment by valyala** [comment]
The issue must be fixed in [v0.23.0-victorialogs](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v0.23.0-victorialogs) release.
